TypeCobol/Compiler/Sql/Scanner/SqlScanner.cs

```diff
@@ -35,6 +48,23 @@ public override Token GetTokenStartingFrom(int startIndex)
 
             // Start scanning at the given index
             currentIndex = startIndex;
+            if (IsSqlKeywordPart(line[currentIndex]))
+            {
+                //Consume all sql-keyword compatible chars
+                for (; currentIndex <= lastIndex && IsSqlKeywordPart(line[currentIndex]); currentIndex++)
+                {
+                }
+
+                string tokenText = line.Substring(startIndex, currentIndex - startIndex);
+                //Try to match keyword text
+                var tokenType =
+                    TokenUtils.GetSqlKeywordTokenTypeFromTokenString(tokenText);
+                if (tokenType!=TokenType.UserDefinedWord)
+                {
+                    return new Token(tokenType, startIndex, currentIndex - 1, tokensLine);
+                }
+            }
```

Comment:
Why is it still here ? We need to start with the main switch, the case keyword or user-defined wword comes last.